### PR TITLE
feat(mcp-server): Add CSS value fuzzy filtering, token format conversion, and percentage validation

### PR DESCRIPTION
**Pull request by mahmoudmoravej**
## Summary

This PR enhances the MCP server with CSS value filtering capabilities, token format conversion utilities, and percentage value validation for width/height properties. These improvements streamline the Figma-to-code workflow and provide better validation for component usage.

### 1. CSS Value Fuzzy Matching (`cssValueMatcher.ts`)
- **Color matching**: Fuzzy hex color matching using RGB distance (configurable threshold, default: 15)
- **Number matching**: Supports rem, px, ms, s, % with configurable tolerances and unit conversion
  - rem ↔ px conversion (1rem = 16px)
  - s ↔ ms conversion (1s = 1000ms)
  - Tolerance-based matching (±2px, ±10ms, ±1%)
- **String matching**: Case-insensitive partial matching for font families, easing functions, etc.
- **Pure numbers**: Exact matching for values like font-weight (400, 500, etc.)

### 2. Token Format Converter (`tokenFormatConverter.ts`)
- Converts between full format (with `cssValue` and `propValue`) and brief format (propValue only)
- Recursive conversion handling nested token structures
- Enables single unified token map files instead of separate brief/full versions
- Reduces file duplication and improves maintainability

### 3. Enhanced `get_design_tokens` Tool
- **New search parameters**:
  - `search_token_names`: Filter by token names (existing functionality, improved)
  - `search_css_values`: Filter by CSS values (e.g., `['#3c3c3c', '1rem', '400']`)
  - `include_css_values`: Toggle between brief/full format response
- **Combined filtering**: Supports both name and CSS value filters simultaneously
- **Consolidated files**: Single token map file per category
- **Performance**: Token data caching to avoid redundant file reads

### 4. Percentage Value Validation
- Added validation rule: `width`, `height`, `maxWidth`, `minWidth`, `maxHeight`, `minHeight` accept percentage values without `UNSAFE_` prefix
- Prevents incorrect usage like `UNSAFE_width="100%"` (should be `width="100%"`)
- Comprehensive error messages guiding users to correct usage
- 21+ new test cases covering valid/invalid usage and edge cases

### 5. Prompt and Documentation Improvements
- Updated Figma code generation prompt with CSS value filtering workflow
- Added warning against using Core tokens when Semantic tokens are available
- Improved token mapping instructions in prompts
- Enhanced test coverage and documentation

## Test Coverage

- **`cssValueMatcher.test.ts`**: 35+ tests covering all matching scenarios
- **`convertToBriefFormat.test.ts`**: 20+ tests for format conversion
- **`docs.test.ts`**: Enhanced with CSS value filtering tests (15+ new tests)
- **`validateComponentStructure.test.ts`**: 21+ tests for percentage validation
- All existing tests updated and passing

## Benefits

1. **Better Figma-to-Code workflow**: Search tokens by CSS values extracted from Figma designs
2. **Reduced file duplication**: Single token map file handles both brief and full formats
3. **Flexible searching**: Fuzzy matching handles slight color variations and numeric differences
4. **Better maintainability**: Centralized token format conversion logic
5. **Improved validation**: Prevents common mistakes with percentage values and UNSAFE_ props
6. **Enhanced developer experience**: Better error messages and validation feedback

## Files Changed

- **New files**: 4 (cssValueMatcher, tokenFormatConverter, + 2 test files)
- **Modified**: 15+ (tools, utilities, tests, prompts, config, validation)
- **Lines**: +1,500/-300 (net: +1,200)

## Breaking Changes

None. All changes are backward compatible - existing token filtering by name continues to work as before.

## Testing

All tests pass:
```bash
pnpm test
```

### CODE REVIEW

File: apps/mcp-server/package.json [hunk 1]
```diff
@@ -51,7 +51,7 @@
     "nodemon": "3.1.10",
     "rehype-remark": "10.0.1",
     "remark": "15.0.1",
-    "ts-jest": "29.2.5",
```
[the remark this answers is not in the document]
Comment (mahmoudmoravej) [reply apps/mcp-server/package.json]: I try to set it up in another PR to avoid mistakes in future.
